### PR TITLE
Show env info in the shell

### PR DESCRIPTION
**Pull request by ericbottard**
Refactor info() command to return a Table rather than its toString().
Add /about bits to the info() output

Fixes https://github.com/spring-cloud/spring-cloud-dataflow/issues/1247

### CODE REVIEW

**Comment by ghillert** [comment]
3 issues:

1)

@ericbottard There seem to be some rendering issues:

image

2)

Also, for better context, I would probably show the `Platform Specific Info` inside the **App Deployer** / **Task Launcher** boxes respectively. 

3)

In the UI I also indicate visually when there are no `Platform Specific Info` available. Right now, the user has no indication.

File: spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/AboutTemplate.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ */
+
+package org.springframework.cloud.dataflow.rest.client;
+
+import org.springframework.cloud.dataflow.rest.resource.about.AboutResource;
+import org.springframework.hateoas.Link;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * Created by ericbottard on 27/03/2017.
```
Comment (ghillert) [inline spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/AboutTemplate.java]: Provide author tag

File: spring-cloud-dataflow-rest-resource/src/main/java/org/springframework/cloud/dataflow/rest/Version.java [hunk 2]
```diff
@@ -26,7 +26,7 @@
 	/**
 	 * Should be incremented each time the API evolves whether it's a breaking change or not.
 	 */
-	public static final int REVISION = 13;
+	public static final int REVISION = 14;
```
Comment (ghillert) [inline spring-cloud-dataflow-rest-resource/src/main/java/org/springframework/cloud/dataflow/rest/Version.java]: Just FYI - we need to watch this, there is a competing PR that would bump this already.

File: spring-cloud-dataflow-shell-core/src/main/java/org/springframework/cloud/dataflow/shell/command/DataFlowTables.java [hunk 3]
```diff
@@ -16,9 +16,16 @@
 
 package org.springframework.cloud.dataflow.shell.command;
```
Comment (ghillert) [inline spring-cloud-dataflow-shell-core/src/main/java/org/springframework/cloud/dataflow/shell/command/DataFlowTables.java]: Update copyright

File: spring-cloud-dataflow-shell-core/src/test/java/org/springframework/cloud/dataflow/shell/command/ConfigCommandTests.java [hunk 4]
```diff
@@ -17,10 +17,13 @@
 package org.springframework.cloud.dataflow.shell.command;
```
Comment (ghillert) [inline spring-cloud-dataflow-shell-core/src/test/java/org/springframework/cloud/dataflow/shell/command/ConfigCommandTests.java]: Update Copyright year

File: spring-cloud-dataflow-shell-core/src/test/java/org/springframework/cloud/dataflow/shell/command/ConfigCommandTests.java [hunk 5]
```diff
@@ -32,14 +35,18 @@
...
 import org.springframework.cloud.dataflow.shell.TargetHolder;
 import org.springframework.cloud.dataflow.shell.config.DataFlowShell;
 import org.springframework.hateoas.Link;
 import org.springframework.http.converter.HttpMessageConverter;
 import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;
 import org.springframework.shell.CommandLine;
+import org.springframework.shell.table.Table;
+import org.springframework.util.FileCopyUtils;
 import org.springframework.web.client.RestClientException;
 import org.springframework.web.client.RestTemplate;
```
Comment (ghillert) [inline spring-cloud-dataflow-shell-core/src/test/java/org/springframework/cloud/dataflow/shell/command/ConfigCommandTests.java]: Update author tag

File: spring-cloud-dataflow-shell-core/src/test/resources/org/springframework/cloud/dataflow/shell/command/ConfigCommandTests-testInfo.txt [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+║         │                  Deployer Name: null                              ║
+║         │           Deployer Spi Version: null                              ║
+║         │                   Java Version: 1.8                               ║
+║   App   │           Platform Api Version: null                              ║
+║Deployer │        Platform Client Version: null                              ║
+║         │          Platform Host Version: null                              ║
+║         │                  Platform Type: null                              ║
+║         │            Spring Boot Version: null                              ║
+║         │                 Spring Version: null                              ║
+╟┈┈┈┈┈┈┈┈┈┼┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈╢
```
Comment (ghillert) [inline spring-cloud-dataflow-shell-core/src/test/resources/org/springframework/cloud/dataflow/shell/command/ConfigCommandTests-testInfo.txt]: Rendering issue

File: spring-cloud-dataflow-shell-core/src/main/java/org/springframework/cloud/dataflow/shell/command/DataFlowTables.java (hunk 3 above)
Comment (ghillert) [inline spring-cloud-dataflow-shell-core/src/main/java/org/springframework/cloud/dataflow/shell/command/DataFlowTables.java]: Update Copyright year

File: spring-cloud-dataflow-shell-core/src/main/java/org/springframework/cloud/dataflow/shell/command/ConfigCommands.java [hunk 7]
```diff
@@ -47,8 +57,17 @@
...
+import org.springframework.shell.table.KeyValueSizeConstraints;
+import org.springframework.shell.table.KeyValueTextWrapper;
+import org.springframework.shell.table.SimpleHorizontalAligner;
+import org.springframework.shell.table.SimpleVerticalAligner;
 import org.springframework.shell.table.TableBuilder;
 import org.springframework.shell.table.TableModelBuilder;
+import org.springframework.shell.table.Tables;
 import org.springframework.stereotype.Component;
 import org.springframework.util.StringUtils;
 import org.springframework.web.client.RestTemplate;
```
Comment (ghillert) [inline spring-cloud-dataflow-shell-core/src/main/java/org/springframework/cloud/dataflow/shell/command/ConfigCommands.java]: Add author tag

File: spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/DataFlowOperations.java [hunk 8]
```diff
@@ -66,7 +66,12 @@
 	CompletionOperations completionOperations();
```
Comment (ghillert) [inline spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/DataFlowOperations.java]: Update Copyright year + author tag

**Comment by ghillert** [comment]
@ericbottard Once you addressed the 3 issues above + made sure that author tags and copyright years are updated, I can merge.

**Comment by ericbottard** [comment]
Regarding the 3 issues:
1) it's very weird but I remember someone also reporting that, while there was really no issue on another machine. Which OS / terminal etc. are you on?
2) due to that rendering issue, you may not realize it, but the extra info is only separated by that thin dashed line, relating it to the deployer/launcher.
3) extra info is not spec'd and could be anything. The lack of it is not important IMO. When I'm using local deployer and there is no say "API endpoint", then I don't care to know that there is no extra info. You don't miss what you're not aware of.

File: spring-cloud-dataflow-shell-core/src/test/resources/org/springframework/cloud/dataflow/shell/command/ConfigCommandTests-testInfo.txt (hunk 6 above)
Comment (ericbottard) [inline spring-cloud-dataflow-shell-core/src/test/resources/org/springframework/cloud/dataflow/shell/command/ConfigCommandTests-testInfo.txt]: where exactly? This is the expected output

**Comment by ericbottard** [comment]
Addressed other review comments, rebased and force pushed

**Comment by ilayaperumalg** [comment]
I tried with rendering `dataflow config info` with `Platform Specific Info` and it looked good at my local.

**Comment by ghillert** [comment]
Merged with commit d102f2c21916e5a4c42aa6705fcc8024240edc9e

**Pull request closed without merging**